eo-maven-plugin/src/main/java/org/eolang/maven/OyRemote.java

```diff
@@ -40,35 +40,84 @@ final class OyRemote implements Objectionary {
     /**
      * The address template.
      */
-    private final String template;
+    private final UrlOy template;
 
     /**
      * Constructor.
      * @param hash Commit hash
      * @throws IOException if fails.
      */
     OyRemote(final CommitHash hash) throws IOException {
-        this.template = String.format(
-            "https://raw.githubusercontent.com/objectionary/home/%s/objects/%%s.eo",
+        this.template = new UrlOy(
+            "https://raw.githubusercontent.com/objectionary/home/%s/objects/%s.eo",
             hash.value()
         );
     }
 
     @Override
     public String toString() {
-        return this.template;
+        return this.template.toString();
     }
 
     @Override
     public Input get(final String name) throws MalformedURLException {
-        final URL url = new URL(
-            String.format(this.template, name.replace(".", "/"))
-        );
+        final URL url = this.template.get(name);
         Logger.debug(
             this, "The object '%s' will be pulled from %s...",
             name, url
         );
         return new InputOf(url);
     }
 
+    /**
+     * Objectionary URL template.
+     *
+     * @since 1.0
+     */
+    public static class UrlOy {
+
+        /**
+         * URL template. Expects two placeholders in terms of
+         * {@link String#format(String, Object...)}: 1st for hash,
+         * 2nd for program name.
+         */
+        private final String template;
+
+        /**
+         * Objects version hash.
+         */
+        private final String hash;
+
+        /**
+         * Ctor.
+         * @param template URL template.
+         * @param hash Version hash.
```

Comment:
@mximp Why is it `Version`?

Reply:
@volodya-lombrozo as it is  a hash of the git version (or tag) we would like to get the objects for.

Reply:
> @volodya-lombrozo as it is a hash of the git version (or tag) we would like to get the objects for.

It's better to put it into `JavaDoc` - it's useful information that clarify much